Code/Framework/AzCore/AzCore/Module/Environment.h

```diff
@@ -242,41 +255,29 @@ namespace AZ
                 memset(&m_value, 0, sizeof(T));
             }
 
-            template <class... Args>
+            template<class... Args>
             void ConstructImpl(const AZStd::false_type& /* AZStd::has_trivial_constructor<T> */, Args&&... args)
             {
                 // Construction of non-trivial types is left up to the type's constructor.
                 new(&m_value) T(AZStd::forward<Args>(args)...);
             }
-
-            void DestructImpl(const AZStd::true_type& /* AZStd::is_trivially_destructible<T> */)
-            {
-                // do nothing
-            }
-
-            void DestructImpl(const AZStd::false_type& /* AZStd::is_trivially_destructible<T> */)
-            {
-                reinterpret_cast<T*>(&m_value)->~T();
-            }
-
-            // Assumes the lock is already held
-            void UnregisterAndDestruct()
+            static void DestructDispatchNoLock(EnvironmentVariableHolderBase *base, DestroyTarget selfDestruct)
             {
-                // if the environment that created us is gone the owner can be null
-                // which means (assuming intermodule allocator) that the variable is still alive
-                // but can't be found as it's not part of any environment.
-                if (m_environmentOwner)
+                auto *self = reinterpret_cast<EnvironmentVariableHolder *>(base);
+                if(selfDestruct==DestroyTarget::Self)
```

Comment:
nitpick: Can you add spaces around the `==` operator ` `if(` parts of the if statement?

That fits coding style of conditional statements in the codebase.
```suggestion
                if (selfDestruct == DestroyTarget::Self)
```

Reply:
addressed in b01d230